### PR TITLE
Pkg/swiftlint

### PR DESCRIPTION
**Pull request by rhousand**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ x ] x86_64-darwin
  - [ ] aarch64-darwin
- [ x ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ x ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ x ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ x ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `tree ./result`
./result
└── bin
    └── swiftlint

Result of `nixpkgs-review` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>swiftlint</li>
  </ul>
</details>

### CODE REVIEW

File: pkgs/development/tools/swiftlint/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    install -m755 -D ./swiftlint-${version}-macos/bin/swiftlint $out/bin/swiftlint
+  '';
+
+  meta = with lib; {
+    description = "A tool to enforce Swift style and conventions";
+    homepage = "https://github.com/realm/SwiftLint";
+    license = licenses.mit;
+    maintainers = with maintainers; [ rhousand ];
+    platforms = platforms.darwin;
+    hydraPlatforms = [];
```
Comment (FliegendeWurst) [inline pkgs/development/tools/swiftlint/default.nix:25]:
```suggestion
    hydraPlatforms = [];
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
Thanks for packaging swiftlint!
Unfortunately I don't have the hardware required to fully review the package.

File: pkgs/development/tools/swiftlint/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  installPhase = ''
+    install -m755 -D ./swiftlint-${version}-macos/bin/swiftlint $out/bin/swiftlint
+  '';
+
+  meta = with lib; {
+    description = "A tool to enforce Swift style and conventions";
+    homepage = "https://github.com/realm/SwiftLint";
+    license = licenses.mit;
+    maintainers = with maintainers; [ rhousand ];
```
Comment (afh) [inline pkgs/development/tools/swiftlint/default.nix:23]: Adding `meta.mainProgram` is very helpful, especially for folks wanting to `nix run nixpkgs#swiftlint`:
```suggestion
    maintainers = with maintainers; [ rhousand ];
    mainProgram = "swiftlint";
```

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for this, @rhousand, please find below a few suggestions for improvement.

If you wanted to go above and beyond please format the package using [`nixfmt`](https://github.com/NixOS/nixfmt).

Hopefully I've been able to explain well the intention behind the suggestions, if not please ask me to clarify 🙂

**Comment by afh** [comment]
Also if you could resolve the merge conflicts and edit this PRs title to `swiftlint: init 0.55.1`

**Comment by DimitarNestorov** [comment]
SwiftLint was added in #318906

**Pull request closed without merging**